File: server/Features/PromoTasks/Models/PromoTaskModel.cs
namespace Unhurd.Api.Models;

public class PromoTask
{
    public string id { get; set; } = Guid.NewGuid().ToString();
    public string accountId { get; set; } = string.Empty;
    public string title { get; set; } = string.Empty;

    public string description { get; set; } = string.Empty;
    public string status { get; set; } = "ToDo"; // ToDo, InProgress, Done
    public DateTime createdAt { get; set; } = DateTime.UtcNow;
}

File: client/src/models/PromoTaskModel.ts
export type PromoTask = {
  id: string;
  accountId: string;
  title: string;
  description: string;
  status: PromoTaskStatus;
  createdAt: string;
};

export type CreatePromoTask = {
  id?: string;
  accountId: string;
  title: string;
  description: string;
  status: PromoTaskStatus;
};

export type PromoTaskStatus = 'ToDo' | 'InProgress' | 'Done';
